pkg/reconcile/pipeline/context/impl.go

```diff
@@ -227,16 +267,20 @@ func (i *impl) persistBinding() error {
 }
 
 func (i *impl) persistSecret() (string, error) {
-	if i.bindingItems == nil || len(i.bindingItems) == 0 {
+	name, secretExist := i.bindingSecretName()
```

Comment:
Perhaps `checkBindingSecretName` would fit better? Another option would be split between the `secretName` getter and the check whether the secret exists in the internal `bindingItems`.

Reply:
the function does not perform a check - it return the binding secret name that can be used in stages for injecting it into proper resources. That name can be computed, or it could be reused from a already existing secret, if not additional bindings need to be added. However, the caller does not need to know that. Hence, the function return the secret name and information if the secret under that name already exist. This is pure internal method, and that information is needed in `Close()` method to decide if the secret need to be created or not.